src/test/run-make-fulldeps/issue-97463-abi-param-passing/bad.c

```diff
@@ -0,0 +1,24 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+
+
+struct bloc {
+    uint16_t a;
+    uint16_t b;
+    uint16_t c;
+};
+
+uint16_t c_read_value(uint32_t a, uint32_t b, uint32_t c) {
```

Comment:
let me go inspect that.

(whatever it is, it will need to use `-O` to replicate the problems here...)

Reply:
`auxiliary/rust_test_helpers.c` doesn't currently include `<stdlib.h>`, which is "needed" here if we want to reproduce the original test faithfully, in that it calls `malloc`. (And given how sensitive reproducing the failure is to small changes, I'm inclined to assume that `malloc` is a necessity.)

@nagisa do you know offhand if we are deliberately trying to avoid introducing a dependence from `rust_test_helpers.c` upon `stdlib.h` and/or `malloc` itself, perhaps to ensure we can compile the file on esoteric targets that do not have that available? Just curious.

I'll keep moving forward with trying to move this test into `auxiliary/rust_test_helpers.c`, under the assumption that we *can* add such a dependence. But given the amount of work I put into getting the codegen tests working, I'm sort of inclined to just live with a run-make test here, if I encounter any hurdles on the way...